client/client.go

```diff
@@ -191,25 +192,32 @@ func loop(c Coordinator, client *http.Client) error {
 
 // decorrelated Jitter increases the maximum jitter based on the last random value.
 type decorrelatedJitter struct {
-	duration float64 // sleep time
-	min      float64 // min sleep time
-	cap      float64 // max sleep time
+	duration time.Duration // sleep time
+	min      time.Duration // min sleep time
+	cap      time.Duration // max sleep time
 }
 
 func newJitter() decorrelatedJitter {
 	rand.Seed(time.Now().UnixNano())
 	return decorrelatedJitter{
-		duration: 1,
-		min:      1,
-		cap:      10,
+		min: *minJitter,
+		cap: *maxJitter,
 	}
 }
 
 func (d *decorrelatedJitter) calc() {
-	d.duration = math.Min(d.cap, d.min+rand.Float64()*(d.duration*3-d.min))
+	change := rand.Float64() * float64(d.duration*time.Duration(3)-d.min)
+	d.duration = d.min + time.Duration(change)
+	if d.duration > d.cap {
+		d.duration = d.cap
+	}
+	if d.duration < d.min {
+		d.duration = d.min
+	}
 }
 
 func (d *decorrelatedJitter) sleep() {
+	d.calc()
```

Comment:
It'd be a bit cleaner to have this return the duration I'd think.

Reply:
I agree, decided not do change it just to reduce changes, but give me a couple of minutes to fix it